# 01 EDA: synthetic churn dataset

Ноутбук фиксирует первичный анализ синтетического датасета для задачи классификации риска оттока клиентов.

In [1]:
from pathlib import Path
import pandas as pd

DATA_PATH = Path('..') / 'data' / 'churn_text_synthetic.csv'
df = pd.read_csv(DATA_PATH)
df.head()

,message,urgency,months_active,support_tickets,discount_requested,channel,segment,churn_intent
0,"сервис стал слишком дорогим, в мобильном прило...",5,28,3,0,phone,individual,1
1,"все работает стабильно, у нескольких сотрудник...",5,61,1,0,email,individual,0
2,"верните деньги за последний месяц, при оплате ...",5,63,2,1,chat,individual,1
3,"постоянные ошибки и зависания, после последнег...",5,55,2,1,chat,individual,1
4,"все работает стабильно, у нескольких сотрудник...",5,56,2,0,phone,enterprise,0


In [2]:
df.info()
df.describe(include='all')

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 8 columns):
 #   Column              Non-Null Count   Dtype
---  ------              --------------   -----
 0   message             100000 non-null  str  
 1   urgency             100000 non-null  int64
 2   months_active       100000 non-null  int64
 3   support_tickets     100000 non-null  int64
 4   discount_requested  100000 non-null  int64
 5   channel             100000 non-null  str  
 6   segment             100000 non-null  str  
 7   churn_intent        100000 non-null  int64
dtypes: int64(5), str(3)
memory usage: 29.8 MB


,message,urgency,months_active,support_tickets,discount_requested,channel,segment,churn_intent
count,100000,100000.000000,100000.000000,100000.000000,100000.000000,100000,100000,100000.000000
unique,22133,NaN,NaN,NaN,NaN,4,3,NaN
top,"спасибо за быстрое решение вопроса, в вечернее...",NaN,NaN,NaN,NaN,chat,individual,NaN
freq,62,NaN,NaN,NaN,NaN,41778,55092,NaN
mean,NaN,2.994070,36.437920,1.468640,0.306410,NaN,NaN,0.421590
std,NaN,1.412662,20.808207,1.444811,0.461004,NaN,NaN,0.493816
min,NaN,1.000000,1.000000,0.000000,0.000000,NaN,NaN,0.000000
25%,NaN,2.000000,18.000000,0.000000,0.000000,NaN,NaN,0.000000
50%,NaN,3.000000,36.000000,1.000000,0.000000,NaN,NaN,0.000000
75%,NaN,4.000000,54.000000,2.000000,1.000000,NaN,NaN,1.000000


In [3]:
df.isna().sum()

message               0
urgency               0
months_active         0
support_tickets       0
discount_requested    0
channel               0
segment               0
churn_intent          0
dtype: int64

In [4]:
df['churn_intent'].value_counts(normalize=True).rename('share')

churn_intent
0    0.57841
1    0.42159
Name: share, dtype: float64

Ключевые наблюдения: пропусков нет, целевой класс умеренно несбалансирован, положительный класс составляет около 42%. Для оценки модели используются ROC-AUC, F1, precision и recall.